app/models/trip.rb

```diff
@@ -33,41 +34,78 @@ class Trip < ApplicationRecord
   default_scope { includes(:points).order('created_at ASC') }
 
   scope :from_to, -> (from_lon, from_lat, to_lon, to_lat) {
-    select('trips.*,
+    # Avoid Trips doublon
+    matching_points = Point.select("DISTINCT ON (point_a.trip_id) point_a.*,
```

Comment:
I guess some factoring could be made between fromto fromonly and toonly. But it's really not mandatory.

Reply:
Yes but I am bit worry that by making it generic it will be less understandable.

Reply:
Yep, your right ! Let's keep it three cases.